### DISCUSSION TITLE
Getting order items in order API

### DISCUSSION THREAD

**Discussion by rashedoz** [saleweaver/python-amazon-sp-api#365] [Q&A] [answered]
Hello, the order API doesn't send the items of that orders. I want to show the orders with their items like in Amazon Seller central in my inventory. Calling the get items for every order will be unwise. Is there any other way to achieve this? 

I saw some suggestions to use the reports API. Do I need to create a report for every request?

**Comment by saleweaver** [comment, marked as answer]
Hi,

yes, Reports are the way to go. 

Ideally, you create a subscription using [create_report_schedule](https://sp-api-docs.saleweaver.com/endpoints/reports/#sp_api.api.Reports.create_report_schedule)

```
Reports().create_report_schedule(
    reportType=ReportType.GET_FLAT_FILE_ORDER_REPORT_DATA_SHIPPING, # or whatever report is best for you
    period=Schedules.MINUTES_30.value,
    nextReportCreationTime="2019-12-10T20:11:24.000Z"
)
```

The report will be generated every 30 minutes (or whatever value is best for you), starting from `nextReportCreationTime`.

To get the reports, you can then request a list of reports every *n* minutes using [get_reports](
https://sp-api-docs.saleweaver.com/endpoints/reports/#sp_api.api.Reports.get_reports), then download the report. 

The better approach is to setup notifications using a queue though. To do this, you can follow this guide to setup a queue: [MWS SQS Notifications](https://docs.developer.amazonservices.com/en_US/subscriptions/Subscriptions_ReceivingNotifications.html). Then register that as a destination for Notifications using [Notifications](https://sp-api-docs.saleweaver.com/endpoints/notifications/)

You can find some examples in this library's tests folder... 

```
from sp_api.api import Notifications, Reports
from sp_api.base import SellingApiException, NotificationType, ReportType, Marketplaces
from sp_api.util import throttle_retry, load_all_pages


def test_create_destination():
    res = Notifications().create_destination(name='test', arn='arn:aws:sqs:us-east-2:444455556666:queue1')
    assert res.payload.get("destinationId") == "TEST_CASE_200_DESTINATION_ID"
    assert res.payload.get("resource").get('sqs').get('arn') == "arn:aws:sqs:us-east-2:444455556666:queue1"


def test_create_subscription():
    res = Notifications().create_subscription(NotificationType.FEED_PROCESSING_FINISHED, destination_id='dest_id')
    assert res.payload.get('destinationId') == 'TEST_CASE_200_DESTINATION_ID'
    assert res.payload.get('subscriptionId') == 'TEST_CASE_200_SUBSCRIPTION_ID'

```

You'll then get a message in the queue you can listen to, and run whatever logic you need only when a report is available.

**Reply by rashedoz** [reply]
Hello, I get the idea here. But it's not ideal for my case. Creating reports and accessing data that way every time will be inefficient. The amazon seller central dashboard shows the data in a convenient way as I need. 

The seller central provides data to their system with necessary order data and order items in a single API. I have also attached an image and object below.
`{
    "amazonOrderId": "XXX-XXXX-XXXX",
    "orderItems": [
        {
            "asin": "XXX",
            "associatedItems": [],
            "billingCountry": null,
            "conditionNote": "XXXX。",
            "conditionSubtype": null,
            "conditionType": "New",
            "customerOrderItemCode": "XX/XX+A==",
            "extendedTitle": "Product Title",
            "giftMessageText": null,
            "giftWrapType": null,
            "harmonizedCode": null,
            "imageSourceSarek": false,
            "imageUrl": "XX",
            "isGift": false,
            "isHeavyOrBulky": false,
            "listingId": null,
            "orderItemId": "XX",
            "productLink": "XX",
            "productName": Product Name",
            "quantityOrdered": 1,
            "quantityShipped": 1,
            "quantityUnShipped": 0,
            "scheduledDeliveryEndDate": null,
            "scheduledDeliveryStartDate": null,
            "scheduledDeliveryTimeZone": null,
            "sellerSku": "xx-xx-xx",
            "signatureRecommended": false,
            "transparencyItem": false,
            "unitPrice": {
                "Amount": 13000,
                "CurrencyCode": "JPY",
                "EncryptedAmount": null
            }
        }
    ],
    "pickupDate": null,
    "pickupReadyDate": null,
    "pluginBadges": [],
    "pluginOrderStatus": null,
    "relativeOrderDate": "17 hours ago",
    "salesChannel": "Amazon.co.jp",
    "sellerOrderId": null,
    "shippingService": "Standard",
    "shippingServiceOption": null,
    "shippingServiceStringId": "shipping-speed-standard_3248",
    "timeZoneDetailsMap": {
        "earliestDeliveryDate": {
            "offset": 32400000,
            "shortenedTzString": "JST"
        },
        "earliestShipDate": {
            "offset": 32400000,
            "shortenedTzString": "JST"
        },
        "latestDeliveryDate": {
            "offset": 32400000,
            "shortenedTzString": "JST"
        },
        "latestShipDate": {
            "offset": 32400000,
            "shortenedTzString": "JST"
        },
        "orderDate": {
            "offset": 32400000,
            "shortenedTzString": "JST"
        }
    }
}`

I want to add a view like this in my system to show the order items and order id in a row. But calling order items api for all orders to get the items is not convinient.
![sc_1](https://user-images.githubusercontent.com/12187788/152218240-f1ee9d0a-4ace-498c-b074-1c78dfc77952.jpg)
